Classes/Format/Alto.php

```diff
@@ -42,4 +46,49 @@ public function getRawText(\SimpleXMLElement $xml)
         }
         return $rawText;
     }
+
+
+
+    /**
+     * This extracts the fulltext data from ALTO XML and returns it in MiniOCR format
+     *
+     * @access public
+     *
+     * @param \SimpleXMLElement $xml: The XML to extract the raw text from
+     *
+     * @return string The unformatted fulltext in MiniOCR format
+     */
+    public function getTextAsMiniOcr(\SimpleXMLElement $xml)
+    {
+        $rawText = '';
+        $xml->registerXPathNamespace('alto', 'http://www.loc.gov/standards/alto/ns-v2#');
+
+        // get all text blocks
+        $blocks = $xml->xpath('./alto:Layout/alto:Page/alto:PrintSpace//alto:TextBlock');
+
+        if (empty($blocks)) {
+            return '';
+        }
+
+        $miniOcr = new \SimpleXMLElement("<ocr></ocr>");
+
+        foreach ($blocks as $block) {
+            $newBlock = $miniOcr->addChild('b');
+            foreach ($block->children() as $key => $value) {
+                if ($key === "TextLine") {
+                    $newLine = $newBlock->addChild('l');
+                    foreach ($value->children() as $lkey => $word) {
+                        if ($lkey == "String") {
+                            $attributes = $word->attributes();
+                            $la =  Solr::escapeQuery((string) $attributes['CONTENT']);
+                            $newWord = $newLine->addChild('w',  htmlspecialchars(Solr::escapeQuery((string) $attributes['CONTENT'])) . ' ');
+                            $newWord->addAttribute('x', (string) $attributes['HPOS'] . ' ' . (string) $attributes['VPOS'] . ' ' . (string) $attributes['WIDTH'] . ' ' . (string) $attributes['HEIGHT']);
+                        }
+                    }
+                }
+            }
+        }
+
+        return $miniOcr->asXml();
```

Comment:
Shouldn't we note that in the case of error of the document creation this method can return false instead of string?

Reply:
Not sure, what the ocr highlighting plugin will accept. I think, we must send an empty string to it. 

Could be improved for sure.

Reply:
But you're right. We shouldn't return the output of `asXml()` directly.